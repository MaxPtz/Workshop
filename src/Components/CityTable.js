import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import './city-table.scss'

export const CityTable = ({ cities }) => {
    if (!Object.keys(cities).length) return null
    return (
        <div className="cities">
            {Object.keys(cities).map((state, i) => (
                <CityRow state={state} cities={cities[state]} nb={i} key={state} />
            ))}
        </div>
    )
}

CityTable.propTypes = {
    cities: PropTypes.object,
}

export const CityRow = ({ state, cities, nb }) => {
    const [show, setShow] = useState(false)

    useEffect(() => {
        setTimeout(() => setShow(true), nb * 400)
        return setShow(false)
    }, [cities])

    return (
        <div className={show ? `city-line show` : `city-line`}>
            <div className="state">{state}</div>
            <div className="state-cities">
                {cities.map((city) => (
                    <>
                        <hr />
                        <div className="city">{city.city}</div>
                    </>
                ))}
            </div>
        </div>
    )
}

CityRow.propTypes = {
    state: PropTypes.string,
    cities: PropTypes.array,
    nb: PropTypes.number,
}
