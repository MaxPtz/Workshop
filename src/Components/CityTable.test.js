import { shallow } from 'enzyme'
import { CityRow, CityTable } from './CityTable'
import React from 'react'

describe('CityTable rendering', () => {
    it('should render CityTable component', function () {
        const mockCities = {
            uk: { state: 'uk', cities: ['London'] },
            belgium: { state: 'belgium', cities: ['bx', 'mons'] },
        }
        expect(shallow(<CityTable cities={mockCities} />).length).toEqual(1)
        expect(shallow(<CityTable cities={mockCities} />).length).toMatchSnapshot()
    })
})

describe('CityRow rendering', () => {
    it('should render CityRow component', function () {
        const mockCities = {
            state: 'uk',
            cities: ['London'],
        }
        expect(
            shallow(
                <CityRow
                    state={mockCities.state}
                    cities={mockCities.cities}
                    nb={mockCities.cities.length}
                />
            ).length
        ).toEqual(1)
        expect(
            shallow(
                <CityRow
                    state={mockCities.state}
                    cities={mockCities.cities}
                    nb={mockCities.cities.length}
                />
            ).length
        ).toMatchSnapshot()
    })
})
