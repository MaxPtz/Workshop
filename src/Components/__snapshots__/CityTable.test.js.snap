// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityRow rendering should render CityRow component 1`] = `1`;

exports[`CityTable rendering should render CityTable component 1`] = `1`;
